File: My Leet Code Java Solutions/Solution1342.java
//1342. Number of Steps to Reduce a Number to Zero
class Solution1342 {
    public int numberOfSteps(int num) {
        int counter = 0;
        while(num!=0){
            if(num % 2==0){
                num/=2;
                counter++;
            }
            else if(num%2==1){
                num -= 1;
                counter++;
            }
        }
        return counter;
    }
}

File: MyLeetCodeSolutions/PythonSolutions/Solution1342.py
#1342. Number of Steps to Reduce a Number to Zero
class Solution1342:
    def numberOfSteps(self, num: int) -> int:
        counter = 0
        while(num!=0):
            if num%2==0:
                num/=2
                counter+=1
            elif num%2==1:
                num-=1
                counter+=1
        return counter